haskell/private/actions/compile.bzl

```diff
@@ -271,7 +271,11 @@ def _compilation_defaults(hs, cc, java, dep_info, srcs, import_dir_map, extra_sr
         ),
     )
 
-def compile_binary(hs, cc, java, dep_info, srcs, ls_modules, import_dir_map, extra_srcs, compiler_flags, dynamic, with_profiling, main_function, version):
+def _hpc_compiler_args(hs):
+    hpcdir = "{}/{}/.hpc".format(hs.bin_dir.path, hs.package_root)
+    return ["-fhpc", "-hpcdir", hpcdir]
+
+def compile_binary(hs, cc, java, dep_info, srcs, ls_modules, import_dir_map, extra_srcs, compiler_flags, dynamic, with_profiling, main_function, version, is_test = False, hpc_outputs = []):
```

Comment:
This would be a nice change, but I think it's out of scope for this ticket now that the PR has been merged -- I can try to incorporate this into later work on coverage. We could also do this for much of the repository all at once, we have so many arguments to a lot of our functions.